src/compiler/code_generator.ts

```diff
@@ -989,17 +965,63 @@ export class CodeGenerator {
         this.functions.push(slot);
         this.target = slot;
         const subCtx: Context = createContext(ctx);
-        this.compileAST(ast.slots[slotName], subCtx);
+        this.compileAST(ast.slots[slotName].content, subCtx);
+        const params = [`__render: ${name}(${ctxStr})`];
+        if (ast.slots[slotName].attrs) {
+          for (const [n, v] of Object.entries(ast.slots[slotName].attrs!)) {
+            params.push(`${n}: ${compileExpr(v) || undefined}`);
+          }
+        }
+        const slotInfo = `{${params.join(", ")}}`;
         if (this.hasRef) {
           slot.code.unshift(`  const refs = ctx.__owl__.refs`);
-          slotStr.push(`'${slotName}': ${name}(${ctxStr})`);
+          slotStr.push(`'${slotName}': ${slotInfo}`);
         } else {
-          slotStr.push(`'${slotName}': ${name}(${ctxStr})`);
+          slotStr.push(`'${slotName}': ${slotInfo}`);
         }
       }
       this.target = initialTarget;
       slotDef = `{${slotStr.join(", ")}}`;
-      extraArgs.slots = slotDef;
+    }
+
+    if (slotDef && !(ast.dynamicProps || hasSlotsProp)) {
+      props.push(`slots: ${slotDef}`);
+    }
+
+    const propStr = `{${props.join(",")}}`;
+
+    let propString = propStr;
+    if (ast.dynamicProps) {
+      if (!props.length) {
+        propString = `${compileExpr(ast.dynamicProps)}`;
+      } else {
+        propString = `Object.assign({}, ${compileExpr(ast.dynamicProps)}, ${propStr})`;
+      }
+    }
+
+    let propVar: string;
+    if ((slotDef && (ast.dynamicProps || hasSlotsProp)) || this.dev) {
```

Comment:
why different in dev mode? i don't think it's important enough to make a difference

Reply:
see if (this.dev) { ... } below

Reply:
indeed :)